src/home/homework.c

```diff
@@ -1912,8 +1912,10 @@ static int home_lock(UserRecord *h) {
                 return r;
         if (r > 0) {
                 r = unit_freezer_freeze(f);
-                if (r < 0)
+                if (r < 0) {
+                        (void) unit_freezer_thaw(f);
```

Comment:
hmm, why this client side? this seems like something the server side should do if necessary.

Reply:
1. The timeout used is scenario-dependent: the value of 5 second (as proposed in this PR) might be appropriate for current in-tree users but might be arbitrary and excessively short for others.
2. Relating to point 1: as the timeout is enforced client-side, we don't actually know if the invocation has actually failed as far as the client is concerned. In the case of what is addressed in https://github.com/systemd/systemd/pull/36778/commits/d4a01637f485c31d875e5a288f8376ea7e774dcb specifically, this will lead to `systemd-sleep` proceeding with sleep anyway and failing, since the freezing is still in progress and haven't yet failed as far as the server is concerned.

For these reasons, I believe that leaving the timeout logic on the client side is the best solution for now.

Reply:
hmm i am not convinced the 5s timeout is really a good choice... what's the rationale for making it this low?

Reply:
Freezing processes is supposed to be very fast (on the order of miliseconds) assuming the processes are only accessing locally stored filesystems. (Note that on most systems NetworkManager will disconnect from the network before going to sleep, causing the freezing of processes accessing a mounted network filesystem to hang, so for this specific case we assume that it will never complete.) Other use cases may have different requirements (e.g. a container manager freezing a container running directly from a network filesystem for checkpointing).

Reply:
we might be on a very busy machine, and there simply be a myriad of processes. i really don't think 5s is anywhere near a good timeout for this...